### ISSUE TITLE
showOnScreen / getOffsetToReveal doesn't take status bar into consideration

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#11663]
This is a problem for a11y. When an item is partly covered by the status bar and a11y focus is moved to it, Android issues a showOnScreen action, which is handled by [1]. It will currently calculate via [2] that the item needs to be aligned with the upper edge of the screen. This is incorrect. It should be aligned with the lower edge of the status bar. 

While this sounds like just a visual annoyance, it can actually cause a11y users to get stuck: When the list is scrolled all the way to the bottom, the showOnScreen action should scroll it down a little. Instead, we currently try to scroll up, but that's not possible as we are already at the end. The user needs to resort to two finger scrolling to get out of this situation. 

[1] https://github.com/flutter/flutter/blob/8f56f6fdd1c6b4586518b927623bc87341452b50/packages/flutter/lib/src/widgets/single_child_scroll_view.dart#L431
[2] https://github.com/flutter/flutter/blob/8f56f6fdd1c6b4586518b927623bc87341452b50/packages/flutter/lib/src/widgets/single_child_scroll_view.dart#L390

**Comment by Hixie** [comment]
We should probably make the "scroll into view" logic optionally take a rect that describes the "view". Then the accessibility code could just exclude any system-level padding when giving it the instruction to scroll.
